src/thor/bidirectional_astar.cc

```diff
@@ -317,13 +317,24 @@ inline bool BidirectionalAStar::ExpandForwardInner(GraphReader& graphreader,
   float sortcost =
       newcost.cost + astarheuristic_forward_.Get(t2->get_node_ll(meta.edge->endnode()), dist);
 
+  InternalTurn turn = InternalTurn::kNoTurn;
```

Comment:
this feels like costing bleeding into the algorithms. is there anyway we can turn this into a function in costing so that the costings that dont care about uturns can do nothing and then motor vehicles can share a base class function that does this?

Reply:
@kevinkreiser lost me a bit here.  please explain

Reply:
@kevinkreiser I am setting `InternalTurn::kLeftTurn` value on the edge label because I need to know if we made a left onto a small internal edge and in the costing we check if this is set and check if we are transitioning off of the internal edge via another left.  This is the type of uturn that we are trying to avoid.  And vice versa for driving on the right.  So I am confused what you want to change to a function.

Reply:
sorry poor suggestion. i see this is only marking what case we have rather than making a decision about it, the costing does that later.

so i guess what im really getting at is, can we encapsulate this into a function on the pathalgorithm class and call this same function from all the different algorithms? it feels bad to copy paste what looks like similar code into every algorithm or am i missing some subtleties?

Reply:
@kevinkreiser so after thinking about this, I realized this morning what you meant.  I will move the following logic to the costing models where it is needed.  

```
  InternalTurn turn = InternalTurn::kNoTurn;
  uint32_t opp_local_idx = pred.opp_local_idx();
  baldr::Turn::Type turntype = meta.edge->turntype(opp_local_idx);
  if (nodeinfo->drive_on_right()) {
    if (meta.edge->internal() && meta.edge->length() <= kShortInternalLength &&
        (turntype == baldr::Turn::Type::kSharpLeft || turntype == baldr::Turn::Type::kLeft))
      turn = InternalTurn::kLeftTurn;
  } else if (meta.edge->internal() && meta.edge->length() <= kShortInternalLength &&
             (turntype == baldr::Turn::Type::kSharpRight || turntype == baldr::Turn::Type::kRight))
    turn = InternalTurn::kRightTurn;
```

Reply:
@kevinkreiser updated.

Reply:
I think it is better in the costing as if we are not interested in this we just return `InternalTurn::kNoTurn` and don't override the TurnType function.  I think in the future we should consider adding additional logic for trucks.  Like what do we consider a short internal edge for trucks vs autos.